### PR TITLE
Implementation of softdouble rounding to int64_t

### PR DESCRIPTION
**Pull request by terfendail**
### This pullrequest changes
Added implementation of softdouble rounding to int64_t and test for rounding and saturation_cast to int and int64

### CODE REVIEW

File: modules/core/test/test_math.cpp [hunk 1]
```diff
@@ -3736,4 +3736,130 @@ TEST(Core_SoftFloat, sincos64)
...
+    struct testvalues
+    {
+        softdouble inVal;
+        int64_t out64;
+        int32_t out32;
+    } *values = (testvalues*)_values;
+
+    for (int i = 0, maxi = sizeof(_values) / sizeof(_values[0]); i < maxi; i++)
+    {
+        ASSERT_EQ(values[i].out64, cvRound64(values[i].inVal));
```
Comment (alalek) [inline modules/core/test/test_math.cpp]: ASSERT_EQ is for fatal errors, like empty input image.
These checks don't look fatal, so please use `EXPECT_EQ` here.

Comment (savuor) [reply modules/core/test/test_math.cpp]: @alalek What about ASSERTS at other SoftFloat tests in the same file? Are they to be replaced by EXPECT_* ?

**Comment by savuor** [comment]
:+1:

File: modules/core/test/test_math.cpp (hunk 1 above)
Comment (alalek) [reply modules/core/test/test_math.cpp]: Rule is the same: If problem is not critical and doesn't break other checks completely and these checks are independent then `EXPECT` is preferable.
But be careful with large cycles (for each pixel from image) - we don't want 10Gb log files of errors: use `ASSERT` or `break` from cycle loop.

**Pull request merged**